Update Grafana datasource type

Gary-H9 opened the pull request:
After working on issue [335](https://github.com/grafana/terraform-provider-grafana/issues/335) I mistakenly added the datasource [type ](https://github.com/grafana/terraform-provider-grafana/pull/340/files#diff-489a9f185703571407b8ec93e9c9fb4043ab01e6fd3b081728b5032facd852b2R324) as `'github'` and not `'grafana-github-datasource'` in the [PR](https://github.com/grafana/terraform-provider-grafana/pull/340). 

This PR seeks to rectify this (github datasources are not created properly as it currently stands) and also rectifies a couple of typos I noted when working on this. 🙏

I will further improve the docs in the near future to include a github example.

Gary-H9 committed:
:wrench: type 'grafana' -> 'grafana-github-datasource'

Gary-H9 committed:
:pencil2: Fix typos

Gary-H9 committed:
Merge branch 'grafana:master' into update-grafana-datasource-type

Gary-H9 committed:
:wrench: Fix github_url -> githubUrl

Gary-H9 committed:
:wrench: Correct access_token -> accessToken

Gary-H9 committed:
:wrench: Correct attrChecks type variable

Gary-H9 committed:
:memo: Update documentation to include github example

Gary-H9 commented:
I've made further changes, some variable names were incorrect:
* `access_token ` -> `accessToken`
* `github_url` -> `githubUrl`
* `secure_json_data `-> `secureJsonData`

I've also added a github datasource example to the documentation.

Gary-H9 commented:
@julienduchesne - could you help me out here? I'm not overly familiar with Go and I'm not sure why I have numerous failing checks.

julienduchesne commented:
You are right about the type being `grafana-github-datasource` but the attributes are bound correctly already:
- `githubUrl`: https://github.com/grafana/grafana-api-golang-client/blob/master/datasource.go#L99
- `accessToken`: https://github.com/grafana/grafana-api-golang-client/blob/master/datasource.go#L256

What you are seeing is just a syntax translation. Terraform takes in snake case but the API takes in camel case. The translation is done in the code

I think just changing `github` for `grafana-github-datasource` in the examples should do the trick

julienduchesne commented on examples/resources/grafana_data_source/resource.tf:
```diff
@@ -50,3 +50,16 @@ resource "grafana_data_source" "stackdriver" {
   }
 }
+
+resource "grafana_data_source" "github" {
```
Since this requires an external plugin, I wouldn't add it as an example here

Gary-H9 replied:
Understood thanks, I'll remove this.

Gary-H9 committed:
:wrench: Correct github_url + secure_json_data

Gary-H9 committed:
:memo: Remove example which depends on external plugin

julienduchesne commented on grafana/resource_data_source_test.go:
```diff
@@ -448,21 +448,21 @@ func TestAccDataSource_basic(t *testing.T) {
 				}
 				secure_json_data {
-					access_token = "token for github"
+					accessToken = "token for github"
```
Here also, `access_token` refers to the Terraform format. So it was already correct

Gary-H9 replied:
Understood thanks again. Latest push corrects this.

Gary-H9 committed:
:wrench: Correct accessToken -> access_token

Gary-H9 commented:
Thanks for your help @julienduchesne. I've removed the erroneous alterations I made, I think I'd just been looking at this too long and had gotten confused. `github_url`, `secure_json_data` and `access_token` are now in the correct format.

I've also removed the addition to the documentation as per your note.

julienduchesne committed:
Save GithubURL and AccessToken
Set correct plugin types in tests

julienduchesne commented:
You were right that it wasn't working @Gary-H9. The issue was that the values weren't being passed to the API model. See https://github.com/grafana/terraform-provider-grafana/pull/422/files#diff-96f3c606e2c6e71c26a6e75f391fc8f850f8200403cc2d652b663f43cbfc8386R665

I fixed it and added another test. Thanks!

Gary-H9 commented:
Thanks @julienduchesne
